integration/querier_test.go

```diff
@@ -234,11 +225,8 @@ func testQuerierWithBlocksStorageRunningInMicroservicesMode(t *testing.T, stream
 			// Make sure the querier is using the bucket index blocks finder.
 			require.NoError(t, querier.WaitSumMetrics(e2e.Greater(0), "cortex_bucket_index_loads_total"))
 
-			comparingFunction := e2e.Equals
-			if streamingEnabled {
-				// Some metrics can be higher when streaming is enabled. The exact number is not deterministic in every case.
-				comparingFunction = e2e.GreaterOrEqual
-			}
+			// Some metrics can be higher when streaming is enabled. The exact number is not deterministic in every case.
+			comparingFunction := e2e.GreaterOrEqual
```

Comment:
This should be deterministic as of https://github.com/grafana/mimir/pull/8039

Reply:
this was me deleting code without reading it. Thanks for pointing it out

Reply:
thanks for pointing this out. I replaced it with `Equals`, than the test locally 4 times and there were no flakes.